src/ucp/wireup/wireup.c

```diff
@@ -1214,25 +1241,13 @@ ucp_wireup_check_config_intersect(ucp_ep_h ep, ucp_ep_config_key_t *new_key,
 
     memcpy(old_dst_rsc_indices, cm_wireup_ep->dst_rsc_indices,
            sizeof(old_dst_rsc_indices));
-    for (lane = 0; lane < new_key->num_lanes; ++lane) {
-        addr_index = addr_indices[lane];
+    ucp_wireup_dst_rsc_indices_fill(ep, new_key, remote_address, addr_indices,
+                                    new_dst_rsc_indices);
 
-        if (lane == ucp_ep_get_cm_lane(ep)) {
-            ucs_assert(addr_index == UINT_MAX);
-            dst_rsc_index = UCP_NULL_RESOURCE;
-        } else {
-            ucs_assert(addr_index != UINT_MAX);
-            ae            = &remote_address->address_list[addr_index];
-            dst_rsc_index = ae->iface_attr.dst_rsc_index;
-            ucs_assert(!ep->worker->context->config.ext.cm_use_all_devices ||
-                       (dst_rsc_index != UCP_NULL_RESOURCE));
-        }
-
-        /* save destination resource index in the CM wireup EP for doing
-         * further intersections, if needed */
-        cm_wireup_ep->dst_rsc_indices[lane] = dst_rsc_index;
-        new_dst_rsc_indices[lane]           = dst_rsc_index;
-    }
+    /* Save destination resource index in the CM wireup EP for doing
+     * further intersections, if needed */
+    memcpy(cm_wireup_ep->dst_rsc_indices, new_dst_rsc_indices,
```

Comment:
can we pass cm_wireup_ep->dst_rsc_indices directly to ucp_wireup_dst_rsc_indices_fill?

Reply:
done